TheMovieDb/Controller/PersonDetailViewController.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  PersonDetailViewController.swift
+//  TheMovieDb
+//
+//  Created by Karla Rubiano on 31/10/21.
+//
+
+import UIKit
+
+final class PersonDetailViewController: UIViewController {
+    
+    var person: Person? {
+        didSet {
+            DispatchQueue.main.async {
+                self.setupUI()
+            }
+        }
+    }
+    var personID: Int?
+    
+    @IBOutlet weak var personName: UILabel!
+   
+    @IBOutlet weak var imagePerson: UIImageView!
+    
+    @IBOutlet weak var textPerson: UITextView!
+    
+    @IBOutlet weak var knownForDepartmentLabel: UILabel!
+    
+    @IBOutlet weak var idLabel: UILabel!
+    
+    @IBOutlet weak var birthdayLabel: UILabel!
+    
+    @IBOutlet weak var deathdayLabel: UILabel!
+    
+    @IBOutlet weak var popularityLabel: UILabel!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        configureUIDetailPerson()
+        detailPersonID()
+        
+    }
+    
+    func detailPersonID() {
+        guard let id = personID else { return }
+        MovieFacade.get(endpoint: .personDetails(id: id)) { [weak self] (response: Result<Person, MovieError>) in
+            guard let self = self else { return }
+            switch response {
+            case.success(let person):
+                self.person = person
+            case .failure(let failureResult):
+                print(failureResult.localizedDescription)
+                self.showErrorAlert()
+            }
+        }
+    }
+    
+    func showErrorAlert() {
+        let errorAlert = UIAlertController(title: Constants.errorAlertTitle, message: Constants.errorAlertMessage, preferredStyle: .alert)
+        errorAlert.addAction(UIAlertAction(title: Constants.errorAlertButton, style: .default))
+        DispatchQueue.main.async {
+            self.present(errorAlert, animated: true)
+        }
+    }
```

Comment:
**DRY** you can make an extension for the UIViewController and make this func only for view controller.

Reply:
Created the extension.